### PR TITLE
Helm chart - mc-xxx-job: change /usr/bin/mc to mc in _helper txt

### PR DESCRIPTION
**Pull request by JasonChen86899**
## Description
when helm install minio, but this installing hung and command is waiting to time out. 

Found due to k8s job: make-xxxx-job and has following log: 
1. /usr/bin/mc is not found 
2. /etc/minio permission denied

## Motivation and Context
1. change /usr/bin/mc to mc 
2. change configPathmc /etc/minio/mc to ./minio/mc

## How to test this PR?
helm install and observe make-xxxx-job  succeed or failed in k8s

## Types of changes
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [X] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [X] Unit tests added/updated
- [X] Internal documentation updated
- [X] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by JasonChen86899** [comment]
@donatello @krisis please review this pr about helm chart

**Comment by harshavardhana** [comment]
Things have changed can you rebase?

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by JasonChen86899** [comment]
> Things have changed can you rebase?

Done.

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_exclamation_mark: [more...](#420b0681-cc7a-49d3-9bee-ac63eb70cbbd) |
| mint-fs.sh | :heavy_exclamation_mark: [more...](#4c0ed08d-1146-474d-80d2-ce08facecd80) |
| mint-erasure.sh | :heavy_exclamation_mark: [more...](#45ec2199-8211-4eb0-ae04-45ad3605aba4) |
| mint-dist-erasure.sh | :heavy_exclamation_mark: [more...](#4566542b-0882-4a88-b463-9e90291bf421) |
| mint-compress-encrypt-dist-erasure.sh | :heavy_exclamation_mark: [more...](#11f78cc7-bdbe-4fca-be22-472813be6bdd) |
| mint-pools.sh | :heavy_exclamation_mark: [more...](#25476693-ba90-473c-a52b-89b76a306ea2) |
<a name="4566542b-0882-4a88-b463-9e90291bf421"></a>
### 16030-ace4301/mint-dist-erasure.sh.xtrace:
```
DISABLED TIME BEING
```
<a name="11f78cc7-bdbe-4fca-be22-472813be6bdd"></a>
### 16030-ace4301/mint-compress-encrypt-dist-erasure.sh.xtrace:
```
DISABLED TIME BEING
```
<a name="25476693-ba90-473c-a52b-89b76a306ea2"></a>
### 16030-ace4301/mint-pools.sh.xtrace:
```
DISABLED TIME BEING
```
<a name="4c0ed08d-1146-474d-80d2-ce08facecd80"></a>
### 16030-ace4301/mint-fs.sh.xtrace:
```
DISABLED TIME BEING
```
<a name="420b0681-cc7a-49d3-9bee-ac63eb70cbbd"></a>
### 16030-ace4301/mint-large-bucket.sh.xtrace:
```
DISABLED TIME BEING
```
<a name="45ec2199-8211-4eb0-ae04-45ad3605aba4"></a>
### 16030-ace4301/mint-erasure.sh.xtrace:
```
DISABLED TIME BEING
```
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :x: [more...](#26570f09-cfd0-4be2-ab6b-86f04da9e137) |
<a name="26570f09-cfd0-4be2-ab6b-86f04da9e137"></a>
### 16030-b0d7e91/mint-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      15.15.15.7:30771
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp 2f9032d453c8:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 1 seconds
(2/14) Running aws-sdk-java tests ... done in 2 seconds
(3/14) Running aws-sdk-php tests ... done in 42 seconds
(4/14) Running aws-sdk-ruby tests ... done in 7 seconds
(5/14) Running awscli tests ... done in 1 minutes and 1 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 15 seconds
(8/14) Running minio-go tests ... done in 49 seconds
(9/14) Running minio-java tests ... FAILED in 28 seconds
{
  "name": "minio-java",
  "function": "addUser()",
  "duration": 336,
  "status": "FAIL",
  "error": "java.lang.RuntimeException: Request failed with response: {\"Code\":\"InternalError\",\"Message\":\"We encountered an internal error, please try again. (Server not initialized, please try again)\",\"Resource\":\"/minio/admin/v3/add-user\",\"RequestId\":\"1739DCFF675093E1\",\"HostId\":\"ff9d1527-06c5-41db-9462-65fd1a022184\"}\n >>> [io.minio.admin.MinioAdminClient.execute(MinioAdminClient.java:206), io.minio.admin.MinioAdminClient.addUser(MinioAdminClient.java:237), TestMinioAdminClient.addUser(TestMinioAdminClient.java:47), TestMinioAdminClient.runAdminTests(TestMinioAdminClient.java:160), FunctionalTest.runTests(FunctionalTest.java:3782), FunctionalTest.runEndpointTests(FunctionalTest.java:3881), FunctionalTest.main(FunctionalTest.java:4002)]"
}
(9/14) Running minio-js tests ... done in 49 seconds
(10/14) Running minio-py tests ... done in 1 minutes and 43 seconds
(11/14) Running s3cmd tests ... done in 15 seconds
(12/14) Running s3select tests ... done in 4 seconds
(13/14) Running versioning tests ... done in 3 minutes and 4 seconds

Executed 13 out of 14 tests successfully.
```
Deleting image on docker hub
Deleting image locally

File: helm/minio/values.yaml [hunk 1]
```diff
@@ -104,7 +104,7 @@ existingSecret: ""
 
 ## Directory on the MinIO pof
 certsPath: "/etc/minio/certs/"
-configPathmc: "/etc/minio/mc/"
+configPathmc: "./minio/mc/"
```
Comment (harshavardhana) [inline helm/minio/values.yaml:107]: This is a breaking change, why is this being changed?

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: helm/minio/values.yaml (hunk 1 above)
Comment (JasonChen86899) [reply helm/minio/values.yaml:107]: > This is a breaking change, why is this being changed?
when mc client container use current system account do mc-job , permission denied will happen in dir `/etc/minio`.  so must be in this system account available  directory to do that shell command, so change it to` ./minio/mc/`

Comment (harshavardhana) [reply helm/minio/values.yaml:107]: such users can configure the mc config path properly, we shouldn't change the default values.

File: helm/minio/templates/_helper_create_bucket.txt [hunk 2]
```diff
@@ -3,9 +3,9 @@ set -e ; # Have script exit in the event of a failed command.
 
 {{- if .Values.configPathmc }}
 MC_CONFIG_DIR="{{ .Values.configPathmc }}"
-MC="/usr/bin/mc --insecure --config-dir ${MC_CONFIG_DIR}"
```
Comment (harshavardhana) [inline helm/minio/templates/_helper_create_bucket.txt:6]: why is this being changed?

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
| mint-erasure.sh | :x: [more...](#a191ec41-2d4f-4778-bde1-ce1dbe6ca5fe) |
<a name="a191ec41-2d4f-4778-bde1-ce1dbe6ca5fe"></a>
### 16030-9344961/mint-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      15.15.15.4:31428
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp 3346fc0103ff:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 7 seconds
(2/14) Running aws-sdk-java tests ... done in 2 seconds
(3/14) Running aws-sdk-php tests ... done in 42 seconds
(4/14) Running aws-sdk-ruby tests ... done in 8 seconds
(5/14) Running awscli tests ... FAILED in 2 minutes and 5 seconds
{
  "name": "awscli",
  "duration": 65203,
  "function": "aws --endpoint-url http://15.15.15.4:31428 s3 sync --no-progress /mint/data s3://awscli-mint-test-bucket-2908/\n",
  "status": "FAIL",
  "error": "upload: ../../../data/datafile-0-b to s3://awscli-mint-test-bucket-2908/datafile-0-bupload: ../../../data/datafile-10-kB to s3://awscli-mint-test-bucket-2908/datafile-10-kBupload: ../../../data/datafile-1-kB to s3://awscli-mint-test-bucket-2908/datafile-1-kBupload: ../../../data/datafile-1-b to s3://awscli-mint-test-bucket-2908/datafile-1-bupload: ../../../data/datafile-33-kB to s3://awscli-mint-test-bucket-2908/datafile-33-kBupload: ../../../data/datafile-100-kB to s3://awscli-mint-test-bucket-2908/datafile-100-kBupload: ../../../data/datafile-1.03-MB to s3://awscli-mint-test-bucket-2908/datafile-1.03-MBupload: ../../../data/datafile-1-MB to s3://awscli-mint-test-bucket-2908/datafile-1-MBupload: ../../../data/datafile-5-MB to s3://awscli-mint-test-bucket-2908/datafile-5-MBupload: ../../../data/datafile-5243880-b to s3://awscli-mint-test-bucket-2908/datafile-5243880-bupload: ../../../data/datafile-6-MB to s3://awscli-mint-test-bucket-2908/datafile-6-MBupload failed: ../../../data/datafile-10-MB to s3://awscli-mint-test-bucket-2908/datafile-10-MB An error occurred (NoSuchUpload) when calling the CompleteMultipartUpload operation: The specified multipart upload does not exist. The upload ID may be invalid, or the upload may have been aborted or completed.upload: ../../../data/datafile-11-MB to s3://awscli-mint-test-bucket-2908/datafile-11-MBupload: ../../../data/datafile-65-MB to s3://awscli-mint-test-bucket-2908/datafile-65-MBupload: ../../../data/datafile-129-MB to s3://awscli-mint-test-bucket-2908/datafile-129-MB"
}
(5/14) Running healthcheck tests ... done in 0 seconds
(6/14) Running mc tests ... done in 15 seconds
(7/14) Running minio-go tests ... done in 35 seconds
(8/14) Running minio-java tests ... done in 18 seconds
(9/14) Running minio-js tests ... done in 46 seconds
(10/14) Running minio-py tests ... done in 1 minutes and 32 seconds
(11/14) Running s3cmd tests ... done in 15 seconds
(12/14) Running s3select tests ... done in 2 seconds
(13/14) Running versioning tests ... done in 3 minutes and 8 seconds

Executed 13 out of 14 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Comment by klauspost** [comment]
Closing, since we cannot take on this change.

**Pull request closed without merging**